mp/src/public/tier1/ilocalize.h

```diff
@@ -319,6 +319,43 @@ class CLocalizedStringArg<float> : public CLocalizedStringArgPrintfImpl<float>
     CLocalizedStringArg( float fValue ) : CLocalizedStringArgPrintfImpl<float>( fValue, LOCCHAR("%.2f") ) {}
 };
 
+// --------------------------------------------------------------------------
+
+#ifdef _WIN32
+#define ___WIDECHAR_PRINT_FORMAT_WIDECHAR LOCCHAR("%s")
+#define ___WIDECHAR_PRINT_FORMAT_ANSICHAR LOCCHAR("%S") // Thanks, microsoft
+#elif defined(POSIX)
+#define ___WIDECHAR_PRINT_FORMAT_WIDECHAR LOCCHAR("%ls")
+#define ___WIDECHAR_PRINT_FORMAT_ANSICHAR LOCCHAR("%s")
+#else
+#warning "ILocalize needs some string macros defined!"
+#endif
+
+template < >
+class CLocalizedStringArg<locchar_t*> : public CLocalizedStringArgPrintfImpl<locchar_t*>
+{
+public:
+	CLocalizedStringArg(locchar_t *pwszValue) : CLocalizedStringArgPrintfImpl<locchar_t*>(pwszValue, ___WIDECHAR_PRINT_FORMAT_WIDECHAR) {}
+};
+
+// --------------------------------------------------------------------------
+
+template < >
+class CLocalizedStringArg<char*> : public CLocalizedStringArgPrintfImpl<char*>
```

Comment:
Also non-const version isn't really needed, it will auto promote to const if necessary